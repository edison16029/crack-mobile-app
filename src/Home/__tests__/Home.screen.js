import React from 'react';
import {TouchableOpacity} from 'react-native';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';

import HomeScreen from '../Home.screen';

const setupHomeScreen = inProps => {
  const props = {
    navigation: {
      navigate: jest.fn(),
    },
    ...inProps,
  };
  const component = shallow(<HomeScreen {...props} />);
  return {
    component,
    props,
  };
};
describe('Home Screen', () => {
  it('should match snapshot', () => {
    const {component} = setupHomeScreen();
    expect(toJson(component)).toMatchSnapshot();
  });

  it('`onPress` should call navigate with value `SinglePlayerGame`', () => {
    const mockNavigate = jest.fn();
    const navigation = {
      navigate: mockNavigate,
    };
    const {component} = setupHomeScreen({navigation: navigation});
    const touchableComponentProps = component.find(TouchableOpacity).at(0).props();
    touchableComponentProps.onPress();
    expect(mockNavigate).toBeCalledTimes(1);
    expect(mockNavigate).toBeCalledWith('SinglePlayerGame');
  });
});
