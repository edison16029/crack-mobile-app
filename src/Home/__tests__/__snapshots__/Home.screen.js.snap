// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Screen should match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ForwardRef
    onPress={[Function]}
    style={
      Object {
        "backgroundColor": "red",
        "height": "50%",
      }
    }
  >
    <Text>
      Hello Eddie
    </Text>
  </ForwardRef>
</View>
`;
